bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -330,6 +331,28 @@ private BigtableOptions resolveLegacyOptions(BigtableOptions options) throws IOE
     return options;
   }
 
+
+  /**
+   * Return options with legacy input options, if any, resolved into currently supported options.
+   */
+  public synchronized BigtableClusterName getClusterName() throws IOException {
+    if (options.getClusterName() == null) {
```

Comment:
We should defer lookup of the cluster until a snapshot operation is performed.  Looking up the cluster is an extra call to an admin method, which we only want to do if it's absolutely necessary.